util/uuid/uuid.go

```diff
@@ -0,0 +1,38 @@
+// Copyright (C) 2019-2022 Algorand, Inc.
+// This file is part of go-algorand
+//
+// go-algorand is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// go-algorand is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with go-algorand.  If not, see <https://www.gnu.org/licenses/>.
+
+package uuid
+
+import (
+	"crypto/rand"
+	"fmt"
+)
+
+// New generates a new random UUID string.
+// The UUID string generated using this method does not conform to the UUID spec and
+// contains completly random bytes.
+func New() string {
+	var buffer [16]byte
+	_, err := rand.Read(buffer[:])
+	if err != nil {
+		panic(fmt.Errorf("unable to randomize buffer in uuid.New() : %w", err))
+	}
+	return fmt.Sprintf("%08x-%04x-%04x-%04x-%012x", int64(buffer[0])+int64(buffer[1])<<8+int64(buffer[2])<<16+int64(buffer[3])<<24,
```

Comment:
If you want to make this conform you could just add before this line:
```
	buffer[6] = (buffer[6] & 0x0f) | 0x40 // Version 4
	buffer[8] = (buffer[8] & 0x3f) | 0x80 // Variant is 10
```
as per https://github.com/google/uuid/blob/master/version4.go#L46-L56

Reply:
also this might be easier to read:
```
hex.EncodeToString(buffer[:4]) + "-" + 
hex.EncodeToString(buffer[4:6]) + "-" + 
hex.EncodeToString(buffer[6:8]) + "-" + 
hex.EncodeToString(buffer[8:10]) + "-" + 
hex.EncodeToString(buffer[10:])
```